lib/facility.rb

```diff
@@ -1,14 +1,188 @@
+      else
+        word.capitalize
+      end
+    end
+    # Join the array of words back into a string
+    title_words.join(' ')
+  end
+
+  def self.exempt_words
+    # List of words that should not be capitalized (states, DMV, road directions, etc.)
+    %w[DMV NY MD OR SE NE NW SW]
+  end
+end
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :website,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  # Initializes a facility
+  def initialize(data)
+    @name = data[:name]
+    @address = data[:address]
+    @phone = data[:phone]
+    @website = data[:website]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+  end
+
+  # Determines which state the facility is in and calls the appropriate method
+  def self.create_facility(state, facility_data)
+    # Switch statement to determine which state the facility is in
+    case state
+      # Calls the appropriate method based on the state
+    when 'OR' # Oregon
+      create_facility_from_oregon(facility_data)
+    when 'NY' # New York
+      create_facility_from_new_york(facility_data)
+    when 'MO' # Missouri
+      create_facility_from_missouri(facility_data)
+    else
+      # Returns an error if the state is not supported
+      raise "Unsupported state: #{state}"
+    end
+  end
+
+  # Creates a facility if the first argument in Facility.create_facility is 'OR'
+  def self.create_facility_from_oregon(facility_data)
+    # Map over the array of facility data and create a new Facility object for each facility
+    facility_data.map do |data|
+      name = Helper.title_case(data[:title])
+      address_json = JSON.parse(data[:location_1][:human_address])
+      street = Helper.title_case(address_json['address'])
+      city = Helper.title_case(address_json['city'])
+      state = address_json['state']
+      zip_code = address_json['zip']
+      phone = data[:phone_number]
+      website = data[:website]
+
+      address = "#{street}, #{city}, #{state}, #{zip_code}" # Concatenate address components
+
+      Facility.new(name: name, address: address, phone: phone, website: website)
+    end
+  end
+
+  # Creates a facility if the first argument in Facility.create_facility is 'NY'
+  def self.create_facility_from_new_york(facility_data)
+    facility_data.map do |data|
+      name = Helper.title_case(data[:office_name])
+      street = Helper.title_case(data[:street_address_line_1])
+      city = Helper.title_case(data[:city])
+      state = data[:state]
+      zip_code = data[:zip_code]
+      phone = data[:public_phone_number]
+      # New York does not provide a website for each facility.
+      # Spoke with Kat who said this was fine. Could also be left as nil.
+      website = 'undefined'
```

Comment:
Yeah, and maybe one day they add a website, so could still do `website = data[:website]` because then if they don't have one, it would just be nil.